fdbserver/DeltaTree.h

```diff
@@ -25,13 +25,27 @@
 #include "fdbclient/FDBTypes.h"
 #include "fdbserver/Knobs.h"
 #include <string.h>
+#include <emmintrin.h>
 
 typedef uint64_t Word;
 // Get the number of prefix bytes that are the same between a and b, up to their common length of cl
 static inline int commonPrefixLength(uint8_t const* ap, uint8_t const* bp, int cl) {
 	int i = 0;
+
+	for (; i + 16 < cl; i += 16) {
```

Comment:
This reads past the end of the strings.  (Probably OK if we can ensure that the memory allocator is padding between us and the end of mapped memory with enough bytes)

Reply:
I don't think we can assume this.

Reply:
Unless I'm really missing something, this doesn't read past the end of strings. It checks that `i + 16 < cl` and reads the bytes from `i` to `i + 16`.